src/langcoder.cc

```diff
@@ -285,28 +282,20 @@ quint32 LangCoder::guessId( const QString & lang )
   return code2toInt( lstr.left( 2 ).toLatin1().data() );
 }
 
-QPair< quint32, quint32 > LangCoder::findIdsForName( QString const & name )
+std::pair< quint32, quint32 > LangCoder::findLangIdPairFromStr( QString const & name )
 {
-  QString nameFolded = "|" + name.toCaseFolded() + "|";
-  QRegExp reg( "[^a-z]([a-z]{2,3})-([a-z]{2,3})[^a-z]" );
-  reg.setMinimal( true );
-  int off = 0;
+  static QRegularExpression reg( "(?=([a-z]{2,3}?)-([a-z]{2,3}))", QRegularExpression::CaseInsensitiveOption );
 
-  while ( reg.indexIn( nameFolded, off ) >= 0 ) {
-    quint32 from = guessId( reg.cap( 1 ) );
-    quint32 to   = guessId( reg.cap( 2 ) );
-    if ( from && to )
-      return QPair< quint32, quint32 >( from, to );
```

Comment:
The return condition of the original code is wrong.

guessId doesn't return 0 when failed 😅